File: JAVA/Hollow Butterfly.java
import java.util.Scanner;

class HollowButterfly {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter No. Of Rows:");
        int rows= sc.nextInt();
        //Half Hollow Pyramid.
        for (int i = 1; i <=rows; i++) {
            for (int j = 1; j <=i; j++) {
                if (j==i || j == 1) {
                    System.out.print("*");
                }else{
                    System.out.print(" ");
                }
            }
            int spaces =2* (rows - i);
            for (int j = 1; j <=spaces; j++) {
                System.out.print(" ");
            }
            for (int j = 1; j <=i; j++) {
                if (j==i || j == 1) {
                    System.out.print("*");
                }else{
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
        //Inverted Half Hollow Pyramid.
        for (int i = rows; i >=1; i--) {
            for (int j = 1; j <=i; j++) {
                if (j==i || j == 1) {
                    System.out.print("*");
                }else{
                    System.out.print(" ");
                }
            }
            int spaces =2* (rows - i);
            for (int j = 1; j <=spaces; j++) {
                System.out.print(" ");
            }
            for (int j = 1; j <=i; j++) {
                if (j==i || j == 1) {
                    System.out.print("*");
                }else{
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
    
}

File: C++/Hollow_Butterfly.cpp
#include<iostream>
using namespace std;

int main()
{
    int rows;
    cout<<"Enter No. Of Rows:";
    cin>>rows;
    for(int i =1 ; i<=rows;i++){
        //Hollow Half Triangle
        for(int j=1 ; j<=i;j++){
            if(j==1||j==i){
                cout<<"*";
            }else{
                cout<<" ";
            }
        }
        //spaces
        int spaces=2*(rows-i);
        for(int j=1;j<=spaces;j++){
            cout<<" ";
        }
        //Hollow Half Triangle
        for(int j=1 ; j<=i;j++){
            if(j==1||j==i){
                cout<<"*";
            }else{
                cout<<" ";
            }
        }
        cout<<endl;
    }
    //Inverted Pyramid
    for(int i =rows ; i>=1;i--){
        //Hollow Half Triangle
        for(int j=1 ; j<=i;j++){
            if(j==1||j==i){
                cout<<"*";
            }else{
                cout<<" ";
            }
        }
        //spaces
        int spaces=2*(rows-i);
        for(int j=1;j<=spaces;j++){
            cout<<" ";
        }
        //Hollow Half Triangle
        for(int j=1 ; j<=i;j++){
            if(j==1||j==i){
                cout<<"*";
            }else{
                cout<<" ";
            }
        }
        cout<<endl;
    }
    
    return 0;
}